tests/testutils/test_functional_testutils.py

```diff
@@ -54,6 +54,22 @@ def test_get_functional_test_files_from_directory() -> None:
     get_functional_test_files_from_directory(DATA_DIRECTORY / "u/_no_issue_here")
 
 
+def test_get_functional_test_files_from_crowded_directory() -> None:
+    """Test that we correctly check the functional test directory structures."""
+    with pytest.raises(AssertionError) as exc_info:
+        get_functional_test_files_from_directory(
+            DATA_DIRECTORY / "m", max_file_per_directory=1
+        )
+    assert exc_info.match("m: 4 when the max is 1")
+    assert exc_info.match("max_overflow: 3 when the max is 1")
+    with pytest.raises(AssertionError) as exc_info:
```

Comment:
Shouldn't this test be with 3 so that the output is different?

Reply:
It tests with ``2`` so that the output is different 😄

Reply:
Let's test with 3 instead so we are sure that we are not actually raising a message each time but actually check the number as well.

Reply:
I think we're doing what you suggest only with 2 instead of 3:
```python
    with pytest.raises(AssertionError) as exc_info:
        get_functional_test_files_from_directory(
            DATA_DIRECTORY / "m", max_file_per_directory=1 #1
        )
    assert exc_info.match("m: 4 when the max is 1")  #1
    assert exc_info.match("max_overflow: 3 when the max is 1")  #1
        get_functional_test_files_from_directory(
            DATA_DIRECTORY / "m", max_file_per_directory=2   #2
        )
    assert exc_info.match("m: 4 when the max is 2")    #2
    assert exc_info.match("max_overflow: 3 when the max is 2")    #2

Reply:
Yeah but are we sure we don't raise `"max_overflow: 3 when the max is 3"`? We are only testing the emitting and `f-stringing` here, not the actual comparison which is the main use of the function we are testing.

Reply:
I finally understood.

Reply:
a955ed8ef